### PR TITLE
Save authentication headers on local storage received after signing in

### PR DESCRIPTION
**Pull request by jdrosales17**
#### Task done:

- [X] Save auth headers in local storage after log in
- [X] Add private and public route components that work with the saved headers
- [X] Add new loading component

---
#### Preview:
Jun-04-2019 12-45-46

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 const App = () => (
   <Router>
-    <div>
-      <Switch>
-        <Route path="/" component={LoginPage} />
-        <Route path="/login" component={LoginPage} />
-      </Switch>
-    </div>
+    <Switch>
+      <PublicRoute path="/login" exact component={LoginPage} />
```
Comment (agch-dev) [inline src/App.js]: didn't you have these routes already defined in a routes object? Use them from there

File: src/reducers/sessionReducer.js [hunk 2]
```diff
@@ -16,12 +16,14 @@ const sessionReducer = (state = initialState, action) => {
       };
     }
     case SIGN_IN_SUCCESS: {
+      const authHeaders = { ...action.headers };
```
Comment (agch-dev) [inline src/reducers/sessionReducer.js]: destructure action

**Review by agch-dev — APPROVED** [review]
🙌 nice work! Tiny comments

File: src/App.js (hunk 1 above)
Comment (jdrosales17) [reply src/App.js]: Yes, I forgot to use them here 😅

**Pull request merged**